iib/workers/tasks/opm_operations.py

```diff
@@ -918,3 +922,254 @@ def verify_operator_exists(
+    """
+    from iib.workers.tasks.utils import run_cmd, set_registry_token
+
+    cmd = [
+        opm_path,
+        'index',
+        'rm',
+        '--generate',
+        '--binary-image',
+        binary_image,
+        '--from-index',
+        from_index,
+        '--operators',
+        ','.join(operators),
+    ]
+
+    log.info(
+        'Generating the database file from an existing database %s and excluding'
+        ' the following operator(s): %s',
+        from_index,
+        ', '.join(operators),
+    )
+
+    with set_registry_token(overwrite_from_index_token, from_index, append=True):
+        run_cmd(cmd, {'cwd': base_dir}, exc_msg='Failed to remove operators from the index image')
+
+
+def deprecate_bundles(
+    bundles: List[str],
+    base_dir: str,
+    binary_image: str,
+    from_index: str,
+    overwrite_target_index_token: Optional[str] = None,
+    container_tool: Optional[str] = None,
+) -> None:
+    """
+    Deprecate the specified bundles from the index image.
+
+    Only Dockerfile is created, no build is performed.
+
+    :param list bundles: pull specifications of bundles to deprecate.
+    :param str base_dir: base directory where operation files will be located.
+    :param str binary_image: binary image to be used by the new index image.
+    :param str from_index: index image, from which the bundles will be deprecated.
+    :param str overwrite_target_index_token: the token used for overwriting the input
+        ``from_index`` image. This is required to use ``overwrite_target_index``.
+        The format of the token must be in the format "user:password".
+    :param str container_tool: the container tool to be used to operate on the index image
+    """
+    from iib.workers.tasks.utils import run_cmd, set_registry_token
+
+    cmd = [
+        opm_path,
+        'index',
+        'deprecatetruncate',
+        '--generate',
+        '--binary-image',
+        binary_image,
+        '--from-index',
+        from_index,
+        '--bundles',
+        ','.join(bundles),
+        '--allow-package-removal',
+    ]
+    if container_tool:
+        cmd.append('--container-tool')
+        cmd.append(container_tool)
+    with set_registry_token(overwrite_target_index_token, from_index):
+        run_cmd(cmd, {'cwd': base_dir}, exc_msg='Failed to deprecate the bundles')
+
+
+def _find_index_version(index_image: str) -> str:
+    """
+    Find the OCP version of the given index image.
+
+    :param str index_image: index_image pull spec
+    :return: version of index image
+    :rtype: str
+
+    """
+    from iib.workers.tasks.utils import get_image_label
+
+    index_image_version = index_image.split(":")[1]
+    if index_image_version.startswith("v4."):
+        return index_image_version
+    else:
+        index_version = get_image_label(index_image, 'com.redhat.index.delivery.version')
+        return index_version
+
+
+def set_opm_verison(from_index: Optional[str] = None):
```

Comment:
Maybe define this function inside `set_opm` for clarity ?

Reply:
I was thinking about that as well, but a separate function was easily testable and IMO separate function gave more readability.

Reply:
well, that had an opposite effect on me and I had to find the decorator function to make sense of this one. Maybe we can move it after the decorator function in the file if you prefer to keep them separate

Reply:
Sure, updated